Sources/Classes/RSMoEngageDestination.swift

```diff
@@ -137,15 +165,15 @@ extension RSMoEngageDestination {
         for (key, value) in traits {
             if value is String || value is NSNumber || value is Date {
                 switch key {
-                case RSKeys.Identify.Traits.email: MoEngage.sharedInstance().setUserEmailID(value as? String)
-                case RSKeys.Identify.Traits.name: MoEngage.sharedInstance().setUserName(value as? String)
-                case RSKeys.Identify.Traits.phone: MoEngage.sharedInstance().setUserMobileNo(value)
-                case RSKeys.Identify.Traits.firstName: MoEngage.sharedInstance().setUserAttribute(value, forKey: USER_ATTRIBUTE_USER_FIRST_NAME)
-                case RSKeys.Identify.Traits.lastName: MoEngage.sharedInstance().setUserLastName(value as? String)
-                case RSKeys.Identify.Traits.gender: MoEngage.sharedInstance().setUserAttribute(value, forKey: USER_ATTRIBUTE_USER_GENDER)
-                case RSKeys.Identify.Traits.birthday: identifyDateUserAttribute(value: value, key: USER_ATTRIBUTE_USER_BDAY)
-                case RSKeys.Identify.Traits.address: MoEngage.sharedInstance().setUserAttribute(value, forKey: RSKeys.Identify.Traits.address)
-                case RSKeys.Identify.Traits.age: MoEngage.sharedInstance().setUserAttribute(value, forKey: RSKeys.Identify.Traits.age)
+                case RSKeys.Identify.Traits.email: MoEngageSDKAnalytics.sharedInstance.setEmailID((value as? String)!)
+                case RSKeys.Identify.Traits.name: MoEngageSDKAnalytics.sharedInstance.setName((value as? String)!)
+                case RSKeys.Identify.Traits.phone: MoEngageSDKAnalytics.sharedInstance.setMobileNumber(value as! String)
+                case RSKeys.Identify.Traits.firstName: MoEngageSDKAnalytics.sharedInstance.setUserAttribute(value, withAttributeName: "USER_ATTRIBUTE_USER_FIRST_NAME") 
+                case RSKeys.Identify.Traits.lastName: MoEngageSDKAnalytics.sharedInstance.setLastName((value as? String)!)
+                case RSKeys.Identify.Traits.gender: MoEngageSDKAnalytics.sharedInstance.setUserAttribute(value, withAttributeName: "USER_ATTRIBUTE_USER_GENDER")
+                case RSKeys.Identify.Traits.birthday: identifyDateUserAttribute(value: value, key: "USER_ATTRIBUTE_USER_BDAY")
```

Comment:
I can see previously `USER_ATTRIBUTE_USER_BDAY` was some constant and now it's some string as you've defined it inside "...". Please use the correct key.

Reply:
I searched for key in previous version. Key was not present in earlier project, kindly share if you have any reference for previous key used

Reply:
Please always refer to the destination doc for such references. [Here](https://developers.moengage.com/hc/en-us/articles/4403905883796-Tracking-user-attributes#01GP396BP8DYSAW3MF9W2VMWVB) you go:
```
MoEngageSDKAnalytics.sharedInstance.setDateOfBirth(userBirthdate)//userBirthdate should be a Date instance
```

Reply:
done